core/src/main/scala/org/apache/spark/metrics/MetricsSystem.scala

```diff
@@ -169,9 +176,12 @@ private[spark] class MetricsSystem private (
   }
 
   def removeSource(source: Source): Unit = {
-    sources -= source
-    val regName = buildRegistryName(source)
-    registry.removeMatching((name: String, _: Metric) => name.startsWith(regName))
+    val index = sources.indexOf(source)
```

Comment:
Do you just want to check if `sources.contains` the source?

Reply:
`sources` is an ArrayBuffer.  if  `sources` really contains the checking one,the index can avoid traversing array twice.